src/vs/workbench/contrib/terminal/browser/terminalInstance.ts

```diff
@@ -1763,25 +1773,43 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
 				// absolute Windows file path
 				if (this._processManager.os === OperatingSystem.Windows && title.match(/^[a-zA-Z]:\\.+\.[a-zA-Z]{1,3}/)) {
 					title = path.win32.parse(title).name;
+					this._sequence = title;
 				}
 				break;
 		}
 
 		// Remove special characters that could mess with rendering
 		title = title.replace(/[\n\r\t]/g, '');
 
-		const didTitleChange = title !== this._title;
-		this._title = title;
-		this._titleSource = eventSource;
-		if (didTitleChange) {
+		title = template(this._configHelper.config.title, {
+			process: this._processName || title,
+			sequence: this._sequence,
+			separator: { label: this._configHelper.config.separator }
+		});
+
+		this.getCwd().then(cwd => {
```

Comment:
The template may not contain the cwd so we want to fetch it optionally. With the property change event this will become sync, but if we ended up keeping this we would want to make setTitle return a Promise.

Reply:
I think that it might be best to handle that in a `setDescription `method